src/simulation/dynamics/stateSpaceEffector/_UnitTest/test_stateSpaceStateEffector_verification.py

```diff
@@ -0,0 +1,233 @@
+
+    scObject = spacecraft.Spacecraft()
+    scObject.ModelTag = "scObject"
+    scObject.hub.mHub = 1e18
+    scObject.hub.IHubPntBc_B = np.array([[1e18, 0.0, 0.0], [0.0, 1.5e18, 0.0], [0.0, 0.0, 2e18]])
+    scObject.hub.r_BcB_B = np.array([0.0, 0.0, 0.0])
+    scObject.hub.r_CN_NInit = np.array([0.0, 0.0, 0.0])
+    scObject.hub.v_CN_NInit = np.array([0.0, 0.0, 0.0])
+    scObject.hub.sigma_BNInit = np.array([0.0, 0.0, 0.0])
+    scObject.hub.omega_BN_BInit = np.array([0.0, 0.0, 0.0])
+
+    stateSpace = stateSpaceStateEffector.StateSpaceStateEffector(7)
+    stateSpace.ModelTag = "stateSpace"
+
+    mass1 = 50
+    mass2 = 30
+    IS1PntSc1_F = np.array([[30, 0.0, 0.0],
+                            [0.0, 40, 0.0],
+                            [0.0, 0.0, 50]])
+    IS2PntSc2_F = np.array([[30, 0.0, 0.0],
+                            [0.0, 40, 0.0],
+                            [0.0, 0.0, 50]])
+    r_FB_B = np.array([0.5, -1.0, 1.0])
+    r_Sc1S1_F = np.array([2, 0, 0])
+    r_S2S1_F = np.array([0, 0, 0])
+    r_Sc2S2_F = np.array([3, 2, 0])
+    yHat_F = np.array([0.0, 1.0, 0.0])
+
+    mass = mass1 + mass2
+    r_Sc2S1_F = r_Sc2S2_F + r_S2S1_F
+    r_ScS1_F = (mass1 * r_Sc1S1_F + mass2 * r_Sc2S1_F) / mass
+    r_Sc1Sc_F = r_Sc1S1_F - r_ScS1_F
+    r_Sc2Sc_F = r_Sc2S2_F + r_S2S1_F - r_ScS1_F
+    rTilde_Sc1S1_F = np.array(rbk.v3Tilde(r_Sc1S1_F))
+    rTilde_Sc2S2_F = np.array(rbk.v3Tilde(r_Sc2S2_F))
+    rTilde_S2S1_F = np.array(rbk.v3Tilde(r_S2S1_F))
+    rTilde_Sc2S1_F = np.array(rbk.v3Tilde(r_Sc2S1_F))
+    rTilde_ScS1_F = np.array(rbk.v3Tilde(r_ScS1_F))
+    rTilde_Sc1Sc_F = np.array(rbk.v3Tilde(r_Sc1Sc_F))
+    rTilde_Sc2Sc_F = np.array(rbk.v3Tilde(r_Sc2Sc_F))
+    ISPntS1_F = IS1PntSc1_F - mass1 * rTilde_Sc1S1_F @ rTilde_Sc1S1_F + IS2PntSc2_F - mass2 * rTilde_Sc2S1_F @ rTilde_Sc2S1_F
+    IS2PntS2_F = IS2PntSc2_F - mass2 * rTilde_Sc2S2_F @ rTilde_Sc2S2_F
+    ISPntSc_F = IS1PntSc1_F - mass1 * rTilde_Sc1Sc_F @ rTilde_Sc1Sc_F + IS2PntSc2_F - mass2 * rTilde_Sc2Sc_F @ rTilde_Sc2Sc_F
+
+    M = np.zeros((7, 7))
+    M[0:3, 0:3] = mass * np.identity(3)
+    M[0:3, 3:6] = - mass * rTilde_ScS1_F
+    M[0:3, 6] = - mass2 * rTilde_Sc2S2_F @ yHat_F
+    M[3:6, 0:3] = mass * rTilde_ScS1_F
+    M[3:6, 3:6] = ISPntS1_F
+    M[3:6, 6] = (IS2PntS2_F - mass2 * rTilde_S2S1_F @ rTilde_Sc2S2_F) @ yHat_F
+    M[6, 0:3] = mass2 * yHat_F.transpose() @ rTilde_Sc2S2_F
+    M[6, 3:6] = yHat_F.transpose() @ (IS2PntS2_F + mass2 * rTilde_Sc2S2_F @ rTilde_S2S1_F)
+    M[6, 6] = yHat_F.transpose() @ IS2PntS2_F @ yHat_F
+
+    stateSpace.mass = mass
+    stateSpace.M = M
+    stateSpace.ISPntSc_F = ISPntSc_F
+    stateSpace.r_FB_B = r_FB_B
+    stateSpace.r_ScS_F = r_ScS1_F
+    stateSpace.dcm_FB = [[0.0, 1.0, 0.0],
+                         [-1.0, 0.0, 0.0],
+                         [0.0, 0.0, 1.0]]
+
+    springFrequency = frequency
+    dampingFrequency = 0.0
+    stateSpace.K = [[springFrequency, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
```

Comment:
Can you make these K's and C's just all slightly different?

Reply:
Done.